pytermgui/markup/parsing.py

```diff
@@ -189,120 +188,191 @@ def tokenize_markup(text: str) -> Iterator[Token]:
         yield PlainToken(text[cursor:length])
 
 
-def tokenize_ansi(  # pylint: disable=too-many-locals, too-many-branches, too-many-statements
-    text: str,
+def _tokenize_ansi_color(
+    params: List[str],
 ) -> Iterator[Token]:
-    """Converts some ANSI-coded text into a stream of tokens.
+    """Convert ANSI color code into a stream of tokens
 
     Args:
-        text: Any valid ANSI-coded text.
+        params: List of parameters given to an SGR
 
     Yields:
-        The generated tokens, in the order they occur within the text.
+        The generated tokens
     """
+    state = None
+    color_code = ""
+    for part in params:
+        if state is None:
+            if part in REVERSE_STYLES:
+                yield StyleToken(REVERSE_STYLES[part])
+                continue
 
-    cursor = 0
+            if part in REVERSE_CLEARERS:
+                yield ClearToken(REVERSE_CLEARERS[part])
+                continue
 
-    for matchobj in RE_ANSI.finditer(text):
-        start, end = matchobj.span()
+            if part in ("38", "48"):
+                state = "COLOR"
```

Comment:
I'd use an Enum for state representations; makes it harder to have a typo lead to undefined behaviour.